tests/e2e_tests/conftest.py

```diff
@@ -0,0 +1,422 @@
+import subprocess
+from typing import Optional, List, Dict, Any, Tuple, Union
+from dataclasses import dataclass, field
+
+import pytest
+import docker
+import docker.models.containers
+import docker.models.networks
+from docker.errors import ImageNotFound, NotFound
+from pathlib import Path
+
+from e2e_tests.utils import wait_for_qdrant_ready
+
+
+class QdrantContainer:
+    """Container info object for Qdrant containers."""
+    
+    def __init__(self, container: docker.models.containers.Container, host: str, name: str, 
+                 http_port: int, grpc_port: Optional[int], compose_project: Optional[str] = None) -> None:
+        self.container = container
+        self.host = host
+        self.name = name
+        self.http_port = http_port
+        self.grpc_port = grpc_port
+        self.compose_project = compose_project
+    
+    def __repr__(self) -> str:
+        return f"QdrantContainer(name='{self.name}', host='{self.host}', http_port={self.http_port}, grpc_port={self.grpc_port})"
+
+
+class QdrantCluster:
+    """Cluster info object for Qdrant clusters."""
+    
+    def __init__(self, leader: QdrantContainer, followers: List[QdrantContainer], 
+                 network: docker.models.networks.Network, network_name: str) -> None:
+        self.leader = leader
+        self.followers = followers
+        self.all_nodes = [leader] + followers
+        self.network = network
+        self.network_name = network_name
+    
+    def __repr__(self) -> str:
+        return f"QdrantCluster(leader={self.leader.name}, followers={len(self.followers)}, network='{self.network_name}')"
+
+
+@dataclass
+class QdrantContainerConfig:
+    """Configuration for creating Qdrant containers with proper typing and defaults."""
```

Comment:
Do we have types like this so that our IDEs suggest better things when writing a test?